docs/src/content/config.ts

```diff
@@ -1,13 +1,28 @@
 import { docsSchema, i18nSchema } from '@astrojs/starlight/schema';
-import { defineCollection, z } from 'astro:content';
+import { defineCollection, reference, z } from 'astro:content';
 
-export const collections = {
-	docs: defineCollection({
-		schema: (ctx) =>
-			docsSchema()(ctx).extend({
-				badge: z.enum(['stable', 'unstable', 'experimental']).optional(),
-				contributor: z.string().optional(),
-			}),
+const contributors = defineCollection({
+	type: 'data',
+	schema: z.object({
+		name: z.string(),
+		twitter: z.string().url().optional(),
+		linkedin: z.string().url().optional(),
+		github: z.string().url().optional(),
 	}),
-	i18n: defineCollection({ type: 'data', schema: i18nSchema() }),
+});
+
+const docs = defineCollection({
+	schema: (ctx) =>
+		docsSchema()(ctx).extend({
+			badge: z.enum(['stable', 'unstable', 'experimental']).optional(),
+			contributor: reference('contributors').optional(),
+		}),
+});
+
+const i18n = defineCollection({ type: 'data', schema: i18nSchema() });
+
+export const collections = {
+	docs: docs,
+	i18n: i18n,
+	contributors: contributors,
```

Comment:
nit: I prefer the [shorthand](https://javascript.info/object#property-value-shorthand), mainly due to [this ES Lint rule](https://eslint.org/docs/latest/rules/object-shorthand). I have no strong opinions on this though.

```suggestion
	docs,
	i18n,
	contributors,
```